pymc3/step_methods/sgmcmc.py

```diff
@@ -139,7 +139,7 @@ def __init__(
         self.batch_size = batch_size
         self.total_size = total_size
         _value_error(
-            total_size != None or batch_size != None,
+            total_size is not None or batch_size != None,
```

Comment:
Funny it misses the same problem a second time on this line.

Reply:
Shouldn't ths type of problem be caught by our existing pylint pre-commit?

Reply:
If not, there are at least a couple established `pre-commit`-compatible tools that do exactly these things, so we would need to know why this tool is better.

Reply:
Their are following points that tells why **DeepSource** is better tool . 

- DeepSource analyzers detects in-depth issues than just surface level issues. 2000+ issue coverage.
- Ability to autofix issues and auto format code.
- DeepSource guarantees less than 5% false positives.
- Better integration and developer experience.